tests/unit/utils/txn_parser/test_get_order_book_changes.py

```diff
@@ -0,0 +1,162 @@
+                        "expiration_time": 1686903224,
+                    }
+                ],
+            },
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_offer_partially_filled_and_filled(self: TestGetOrderBookChanges):
+        actual = get_order_book_changes(offer_partially_filled_and_filled["meta"])
+        expected = [
+            {
+                "account": "rNzgS71DyJPMnWMA8aS7NqvXP7bNuwyaZo",
+                "offer_changes": [
+                    {
+                        "direction": "sell",
+                        "quantity": {
+                            "currency": "USD",
+                            "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+                            "value": "-63.7479881398749",
+                        },
+                        "total_price": {
+                            "currency": "USD",
+                            "issuer": "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+                            "value": "-62.4730283770749",
+                        },
+                        "sequence": 5931,
+                        "status": "filled",
+                        "maker_exchange_rate": "0.98",
+                    }
+                ],
+            },
+            {
+                "account": "rPu2feBaViWGmWJhvaF5yLocTVD8FUxd2A",
+                "offer_changes": [
+                    {
+                        "direction": "sell",
+                        "quantity": {
+                            "currency": "USD",
+                            "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+                            "value": "-117.3895136925395",
+                        },
+                        "total_price": {
+                            "currency": "USD",
+                            "issuer": "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+                            "value": "-115.0877585220975",
+                        },
+                        "sequence": 67701941,
+                        "status": "partially-filled",
+                        "maker_exchange_rate": "0.9803921568627452",
+                    }
+                ],
+            },
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_offer_cancelled(self: TestGetOrderBookChanges):
+        actual = get_order_book_changes(offer_cancelled["meta"])
+        expected = [
+            {
+                "account": "rEUt5Wy44vDKBDaGkUWG6oSTvxmqgnKWCg",
+                "offer_changes": [
+                    {
+                        "direction": "buy",
+                        "quantity": {
+                            "currency": "XRP",
+                            "value": "47.504858",
+                        },
+                        "total_price": {
+                            "currency": "XDX",
+                            "issuer": "rMJAXYsbNzhwp7FfYnAsYP5ty3R9XnurPo",
+                            "value": "82335.52909",
+                        },
```

Comment:
(Maybe I'm just getting confused by the names, and `quantity` and `total_price` are meant to be tied directly to the Offer object instead of the net result to the order book?)

Reply:
My reference called it `quantity` and `total_price`. When I think about it, it might be easier and less confusing when the amounts are just named `TakerGets` and `TakerPays`. Also I find the `direction` field a bit confusing. IMO it would make more sense to just return the `flags` field as it is. What do you think?
You're right the amount should always be negative if the `status` is not "created".

Reply:
Since the function is intended to parse order book CHANGES it might be even better to neither return `direction` or `flags` because the fields cannot change.